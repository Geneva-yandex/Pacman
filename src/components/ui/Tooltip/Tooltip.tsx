import * as React from 'react';
import ReactTooltip, {TooltipProps} from 'react-tooltip';
import bem from 'easy-bem';
import './Tooltip.scss';

const b = bem('Tooltip');

function Tooltip(props: React.PropsWithChildren<{ id: string, tooltip: string } & TooltipProps>) {
    const {id, tooltip, children} = props;
    return <div className={b()} data-tip data-for={id}>
        {children}
        <ReactTooltip {...props} effect="solid" id={id}>{tooltip}</ReactTooltip>
    </div>;
}

export default Tooltip;
