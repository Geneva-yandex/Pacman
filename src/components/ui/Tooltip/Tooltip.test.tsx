import * as React from 'react';
import renderer from 'react-test-renderer';
import Tooltip from './Tooltip';

const props = {
    children: 'Tooltip link',
    id: 'test',
    tooltip: 'Tooltip content'
};
describe('Tooltip', () => {
    test('Tooltip component renders correctly', () => {
        const renderedComponent = renderer
            .create(<Tooltip {...props}/>)
            .toJSON();

        expect(renderedComponent).toMatchSnapshot();
    });
});
