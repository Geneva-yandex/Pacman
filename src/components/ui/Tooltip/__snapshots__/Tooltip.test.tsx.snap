// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tooltip Tooltip component renders correctly 1`] = `
<div
  className="Tooltip"
  data-for="test"
  data-tip={true}
>
  Tooltip link
  Tooltip content
</div>
`;
